wgpu-core/src/present.rs

```diff
@@ -34,16 +41,16 @@ pub const DESIRED_NUM_FRAMES: u32 = 3;
 
 #[derive(Debug)]
 pub(crate) struct Presentation {
-    pub(crate) device_id: Stored<DeviceId>,
+    pub(crate) device_id: Valid<DeviceId>,
```

Comment:
Similarly, I would expect `Presentation` to still need a strong reference to its `Device`.

This case is a bit tricky, as `Presentation` isn't specific to any `HalApi` type.